archivessnake/edit_notes.py

```diff
@@ -7,64 +7,67 @@
 import time
 
 from asnake.aspace import ASpace
+from asnake.utils import walk_tree
 #from fuzzywuzzy import fuzz
 from rapidfuzz import fuzz
 
-AS_REPOSITORY = 2
 OUTPUT_FILENAME = "out.csv"
 NOTE_TYPE_CHOICES = ["bioghist", "accessrestrict", "odd", "abstract", "arrangement", "userestrict", "fileplan", "acqinfo", "langmaterial", "physdesc", "prefercite", "processinfo", "relatedmaterial", "separatedmaterial"]
 ACTION_CHOICES = ["modify", "delete"]
 LEVEL = ["collection", "file", "series", "item", "all"]
 CONFIDENCE_RATIO = 97
-spreadsheet_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), OUTPUT_FILENAME)
 
-def process_tree(args, resource):
+def process_tree(client, parser, writer):
     """Iterates through a given collection, file, or series provided by user input. Finds note content that matches user input and then deletes or modifies relevant notes according to user preference.  """
-    for record in resource.tree.walk:
+    args = parser.parse_args()
+    resource_id = client.get(f'/repositories/2/resources/{(args.resource_id)}').json()
```

Comment:
This variable is a bit misleading (especially since one of your args has the same name) because its really the full json of the resource, not the ID.